runtime/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+    #[doc(alias = "IRDescriptorTableSetSampler")]
+    #[allow(unused_variables, unreachable_code)]
+    pub fn sampler(argument: &metal::SamplerState, lod_bias: f32) -> Self {
+        Self {
+            gpuVA: todo!("Add gpu_resource_id() to SamplerState: https://github.com/gfx-rs/metal-rs/pull/328"), // argument.gpu_resource_id()._impl,
+            textureViewID: 0,
+            metadata: lod_bias.to_bits() as u64,
+        }
+    }
+
+    /// Encode an acceleration structure in this descriptor.
+    ///
+    /// This function is a port of the `IRDescriptorTableSetAccelerationStructure` function in the `ir_raytracing.h` header.
+    /// See <https://developer.apple.com/metal/shader-converter/> for more info.
+    #[doc(alias = "IRDescriptorTableSetAccelerationStructure")]
+    pub fn acceleration_structure(gpu_address: u64) -> Self {
+        Self {
+            gpuVA: gpu_address,
+            textureViewID: 0,
+            metadata: 0,
+        }
+    }
+
+    /// Get the metadata value for a buffer view.
+    ///
+    /// This function is a port of the `IRDescriptorTableGetBufferMetadata` function in the `metal_irconverter_runtime.h` header.
+    /// See <https://developer.apple.com/metal/shader-converter/> for more info.
+    pub fn buffer_metadata(view: &BufferView<'_>) -> u64 {
+        let mut metadata = (view.buffer_size & ffi::kIRBufSizeMask) << ffi::kIRBufSizeOffset;
+        metadata |= (view.texture_view_offset_in_elements as u64 & ffi::kIRTexViewMask)
+            << ffi::kIRTexViewOffset;
+        if view.typed_buffer {
+            metadata |= 1 << ffi::kIRTypedBufferOffset;
+        }
+        metadata
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[non_exhaustive]
+#[repr(u64)]
+pub enum BindPoint {
+    #[doc(alias = "kIRRuntimeTessellatorTablesBindPoint")]
+    RuntimeTessellatorTables = ffi::kIRRuntimeTessellatorTablesBindPoint,
+    #[doc(alias = "kIRRayDispatchArgumentsBindPoint")]
+    RayDispatchArguments = ffi::kIRRayDispatchArgumentsBindPoint,
+    #[doc(alias = "kIRArgumentBufferBindPoint")]
+    ArgumentBuffer = ffi::kIRArgumentBufferBindPoint,
+    // TODO: Duplicate with RayDispatch, we shouldn't bother with an enum TBH.
+    // #[doc(alias = "kIRArgumentBufferHullDomainBindPoint")]
+    // ArgumentBufferHullDomain = ffi::kIRArgumentBufferHullDomainBindPoint,
+    #[doc(alias = "kIRDescriptorHeapBindPoint")]
+    DescriptorHeap = ffi::kIRDescriptorHeapBindPoint,
+    #[doc(alias = "kIRSamplerHeapBindPoint")]
+    SamplerHeap = ffi::kIRSamplerHeapBindPoint,
+    #[doc(alias = "kIRArgumentBufferDrawArgumentsBindPoint")]
+    ArgumentBufferDrawArguments = ffi::kIRArgumentBufferDrawArgumentsBindPoint,
+    #[doc(alias = "kIRArgumentBufferUniformsBindPoint")]
+    ArgumentBufferUniforms = ffi::kIRArgumentBufferUniformsBindPoint,
+    #[doc(alias = "kIRVertexBufferBindPoint")]
+    VertexBuffer = ffi::kIRVertexBufferBindPoint,
+}
```

Comment:
We used to have these constants copy-pasted before. That's probably wrong, as their value is only defined _inside the library_. They should be referenced via `const extern kIR...BindPoint`, in case the library changes.